lib/Config.js

```diff
@@ -51,6 +53,42 @@ function restEndpointsAssert(restEndpoints, locationConstraints) {
         'bad config: rest endpoint target not in locationConstraints');
 }
 
+function azureLocationConstraintAssert(location, locationObj) {
+    const storageAccountNameFromEnv =
+        process.env[`${location}_AZURE_STORAGE_ACCOUNT_NAME`];
+    const storageAccessKeyFromEnv =
+        process.env[`${location}_AZURE_STORAGE_ACCESS_KEY`];
+    const { azureStorageAccountName, azureStorageAccessKey,
+        azureContainerName } = locationObj.details;
+    assert(storageAccountNameFromEnv || azureStorageAccountName,
+        'bad location constraint: ' +
+        `"${location}" azureStorageAccountName must be ` +
+        'set in locationConfig or environment variable');
+    assert(storageAccessKeyFromEnv || azureStorageAccessKey,
+        'bad location constraint: ' +
+        `"${location}" azureStorageAccessKey must be ` +
+        'set in locationConfig or environment variable');
+    [storageAccountNameFromEnv, azureStorageAccountName].forEach(value => {
+        if (value !== undefined) {
+            assert(azureAccountNameRegex.test(value),
+                `bad location constraint: "${location}" ` +
+                `azureStorageAccountName "${value}" is an invalid value`);
+        }
+    });
+    [storageAccessKeyFromEnv, azureStorageAccessKey].forEach(value => {
+        if (value !== undefined) {
+            assert(azureAccessKeyRegex.test(value),
+                `bad location constraint: "${location}" ` +
+                'azureStorageAccessKey is not a valid base64 string');
+        }
+    });
+    assert(azureContainerName, 'bad location constraint: ' +
+        `"${location}" azureContainerName must be defined`);
```

Comment:
This would assert false if `locationObj.details.azureContainerName` was defined as something falsy. I don't know why someone would set their config that way, but I guess it's possible.

Reply:
Made it more explicit to check if it is actually undefined and added some checks to make sure it is a string. The isValidBucketName function will check for length (make sure it's not an empty string)